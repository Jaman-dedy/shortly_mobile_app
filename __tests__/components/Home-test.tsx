import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup} from '@testing-library/react-native';

import HomeComponent from '../../src/components/Home';

import MockThemes from '../../mock/MockTheme';

afterEach(cleanup);
test('renders correctly', async () => {
  const tree = await renderer
    .create(
      <MockThemes>
        <HomeComponent />
      </MockThemes>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
