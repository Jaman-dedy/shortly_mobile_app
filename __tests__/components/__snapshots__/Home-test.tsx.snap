// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F0F1F6",
          "flex": 3,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#35323E",
            "fontFamily": "Poppins-Medium",
            "fontSize": 34,
            "fontWeight": "bold",
            "paddingTop": 22,
            "textAlign": "center",
          },
        ]
      }
    >
      Shortly
    </Text>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      < />
      <View
        style={
          Array [
            Object {
              "paddingBottom": 8,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#35323E",
                "fontFamily": "Poppins-Medium",
                "fontSize": 22,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          Let's get started
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#35323E",
                "fontFamily": "Poppins-Medium",
                "fontSize": 18,
                "textAlign": "center",
              },
            ]
          }
        >
          Paste your first link into
the field to shorten it
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#3B3054",
          "flex": 1,
        },
      ]
    }
  >
    <
      style={
        Object {
          "alignSelf": "flex-end",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "paddingTop": 22,
            "position": "absolute",
          },
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Shorten a link here..."
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "white",
            "borderRadius": 4,
            "borderWidth": 1,
            "color": "black",
            "fontSize": 18,
            "height": 45,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
            "textAlign": "center",
            "width": 280,
          }
        }
        testID="text-input"
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2ACFCF",
            "borderRadiu": 4,
            "marginTop": 8,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
            "width": 280,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            SHORTEN IT!
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
